include/zephyr/usb/usbd_msg.h

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief USB support message types and structure
+ */
+
+#ifndef ZEPHYR_INCLUDE_USBD_MSG_H_
+#define ZEPHYR_INCLUDE_USBD_MSG_H_
+
+#include <stdint.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @defgroup usbd_msg_api USB device core API
+ * @ingroup usb
+ * @{
+ */
+
+/**
+ * @brief USB device support message types
+ *
+ * The first set of message types map to event types from the UDC driver API,
```

Comment:
Silently casting the enum types and hoping that numerical value will match is bad design in my opinion. Things will break (because one enum will be updated without updating the other) during development and it'll be hard to track.

Reply:
Right, and it does not make the code smaller.

Reply:
You can update the comment and remove "reserved types are not used"